### PR TITLE
Fusetools2 1093 fix flakiness second iteration

### PR DESCRIPTION
**Pull request by apupier**
based on https://github.com/camel-tooling/vscode-camelk/pull/744

### CODE REVIEW

**Comment by apupier** [comment]
got 3 failures out of 9 builds with current state

**Comment by apupier** [comment]
1 failure out of 10 build.

```
      ✓ Check can deploy Kotlin example: 31123ms

command called: /home/circleci/.config/Code/User/globalStorage/redhat.vscode-camelk/camelk/tools/kamel/kamel with arguments delete,test-basic-kotlin-deploy
Removed the child process running in the background for test-basic-kotlin-deploy: false

      1) "after each" hook for "Check can deploy Kotlin example"

  Suite duration: 179.655 s, Tests: 7
  ```
  
  ```
    1) Check can deploy default examples
       "after each" hook for "Check can deploy Kotlin example":
     Error: Timeout of 20000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/circleci/project/out/src/test/suite/StartIntegration.test.js)
  	at listOnTimeout (internal/timers.js:549:17)
  	at processTimers (internal/timers.js:492:7)
  ```

**Comment by apupier** [comment]
more prevcise error with latest version:

```
command called: /home/circleci/.config/Code/User/globalStorage/redhat.vscode-camelk/camelk/tools/kamel/kamel with arguments run,TestBasicJavaDeploy.java

      ✓ Check can deploy Java example: 2995ms

(node:21051) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 error listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit

(node:21051) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 unpipe listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit
(node:21051) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 error listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit
(node:21051) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit

(node:21051) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 finish listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit

(node:21051) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 unpipe listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit

(node:21051) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit

(node:21051) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 finish listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit

command called: /home/circleci/.config/Code/User/globalStorage/redhat.vscode-camelk/camelk/tools/kamel/kamel with arguments delete,test-basic-java-deploy

Removed the child process running in the background for test-basic-java-deploy: false

      1) "after each" hook for "Check can deploy Java example"

  Suite duration: 31.201 s, Tests: 2
  ```


```
  1 failing

  1) Check can deploy default examples
       "after each" hook for "Check can deploy Java example":
     Error: Undeployment has still not been finished or the Tree view has not been refreshed.
  	at /home/circleci/project/out/src/test/suite/StartIntegration.test.js:120:23
  	at Generator.throw (<anonymous>)
  	at rejected (out/src/test/suite/StartIntegration.test.js:22:65)
```

**Comment by apupier** [comment]
with mor einformation:

```
      ✓ Check can deploy Kotlin example: 30244ms

command called: /home/circleci/.config/Code/User/globalStorage/redhat.vscode-camelk/camelk/tools/kamel/kamel with arguments delete,test-basic-kotlin-deploy
Removed the child process running in the background for test-basic-kotlin-deploy: false

Error while trying to undeploy, try to print more information on the running integrations:

NAME	PHASE	KIT

on the log of integration:

      1) "after each" hook for "Check can deploy Kotlin example"
```
we can suppose that the problem lies in the CamelKNodeProvider which is not updated correctly given thatt direct call to kamel shows that there no more integrations deployed.

**Comment by apupier** [comment]
a single line is responsible to fix the flakiness but the other modificatiosn are provide mor eprecise information and improve readability (which allowed to spot the actual probelm). I think it makes sense to commit altogether also to have the commit history but it is possible to split them in different PRs.

**Comment by apupier** [comment]
we are at 13 out 13 successful build. launched 7 others to be sure.

File: src/test/suite/StartIntegration.test.ts [hunk 1]
```diff
@@ -118,6 +114,40 @@ suite('Check can deploy default examples', () => {
...
+async function cleanDeployedIntegration() {
+	let deployedTreeNode: TreeNode | undefined = await retrieveDeployedTreeNode();
+	if (deployedTreeNode) {
+		await vscode.commands.executeCommand('camelk.integrations.remove', deployedTreeNode);
+		try {
+			await waitUntil(() => {
+				return getCamelKIntegrationsProvider().getTreeNodes().length === 0;
+			}, UNDEPLOY_TIMEOUT);
+		} catch (error) {
+			console.log('Error while trying to undeploy, try to print more information on the running integrations:');
```
Comment (bfitzpat) [inline src/test/suite/StartIntegration.test.ts:126]: I might phrase this a little differently. "Error while trying to remove deployed integration:" I think it's obvious that you have additional details, so you don't need to say that you're trying to print them.

**Comment by apupier** [comment]
21 out of 21 builds in success!

**Review by bfitzpat — APPROVED** [review]
Flaky tests are tough to diagnose. Good work on this one!

**Pull request merged**